src/go/rdctl/pkg/runner/queue_test.go

```diff
@@ -0,0 +1,97 @@
+package runner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+)
+
+func TestTaskRunner(t *testing.T) {
+	t.Run("should run all functions if context not cancelled and no errors", func(t *testing.T) {
+		ctx := context.Background()
+		taskRunner := NewTaskRunner(ctx)
+		ranSlice := []bool{false, false, false}
+		for i := range ranSlice {
+			i := i
+			taskRunner.Add(func() error {
+				ranSlice[i] = true
+				return nil
+			})
+		}
+		if err := taskRunner.Wait(); err != nil {
+			t.Fatalf("unexpected error waiting on taskRunner: %s", err)
+		}
+		for i := range ranSlice {
+			if !ranSlice[i] {
+				t.Errorf("function %d appears to not have run", i)
```

Comment:
Should there be something to test the order the functions were run in?

Reply:
I modified the first test case to test this.